src/main/java/carpet/mixins/BlockStateBase_customStickyMixin.java

```diff
@@ -0,0 +1,31 @@
+package carpet.mixins;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+
+import carpet.fakes.BlockBehaviourInterface;
+import carpet.fakes.BlockStateBaseInterface;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockBehaviour.BlockStateBase;
+import net.minecraft.world.level.block.state.BlockState;
+
+@Mixin(BlockStateBase.class)
+public class BlockStateBase_customStickyMixin implements BlockStateBaseInterface {
+
+    @Shadow private Block getBlock() { return null; }
+    @Shadow private BlockState asState() { return null; }
+
+    @Override
+    public boolean isSticky() {
+        return ((BlockBehaviourInterface)getBlock()).isSticky(asState());
+    }
+
+    @Override
+    public boolean isStickyToNeighbor(Level level, BlockPos pos, BlockPos neighborPos, BlockState neighborState, Direction dir, Direction moveDir) {
```

Comment:
I know that technically accessing blocks is customary to be done via BlockState, but in this case you only use it as a proxy to get to your block information. Seems you might just use (BlockBehaviourInterface)state.getBlock() to get that info, right?

Reply:
That's a fair point. It could be removed without issue.